### PR TITLE
[GEOS-9395] restconfig module for WMTS service configuration

### PR DESCRIPTION
**Pull request by groldan**
Core module (`restconfig-wmts`) to enable configuring basic WMTS service metadata (global and workspace-local) through the REST API just like it's done for WMS, WFS, and WCS.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: doc/en/user/source/rest/api/services.rst [hunk 1]
```diff
@@ -215,4 +215,74 @@ Controls Web Map Service settings for a given workspace.
      -
      -
 
+``/services/wmts/settings[.<format>]``
```
Comment (jodygarnett) [inline doc/en/user/source/rest/api/services.rst]: I am not sure this page needs to be maintained, idea was to replace it with the Open API documentation. We just have not removed these as we are not 100% sure we captured everything.

File: doc/en/api/1.0.0/owsservices.yaml [hunk 2]
```diff
@@ -980,6 +980,233 @@ paths:
         200:
           description: OK
 
+  /services/wmts/settings:
```
Comment (jodygarnett) [inline doc/en/api/1.0.0/owsservices.yaml]: Q: Long term should this file be broken up into multiple individual files?

File: src/restconfig-wmts/src/test/java/org/geoserver/rest/service/LocalWMTSSettingsControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                        .getStatus());
+        boolean thrown = false;
+        try {
+            JSON json =
+                    getAsJSON(
+                            RestBaseController.ROOT_PATH
+                                    + "/services/wmts/workspaces/sf/settings.json");
+        } catch (JSONException e) {
+            thrown = true;
+        }
```
Comment (jodygarnett) [inline src/restconfig-wmts/src/test/java/org/geoserver/rest/service/LocalWMTSSettingsControllerTest.java]: For a shorter test consider:
```
@Test(expected = JSONException.class)
```

Comment (groldan) [reply src/restconfig-wmts/src/test/java/org/geoserver/rest/service/LocalWMTSSettingsControllerTest.java]: right, mind, it's a copy of the other module... maybe we could do some general cleanup but not as part of this pr? do you see this as a blocker?

File: doc/en/user/source/rest/api/services.rst (hunk 1 above)
Comment (groldan) [reply doc/en/user/source/rest/api/services.rst]: well in any case it's now just as useless as the other ones

File: doc/en/api/1.0.0/owsservices.yaml (hunk 2 above)
Comment (groldan) [reply doc/en/api/1.0.0/owsservices.yaml]: Longer term we should provide OpenAPI 3 yaml files, and use tags to split the api into sub-apis, so it wouldn't matter on which files each path is. But would still probably keep these ones on the same file

**Pull request merged**